apps/admin-gui/src/app/shared/components/dialogs/sponsor-this-member-dialog/sponsor-this-member-dialog.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import {
+  NotificatorService,
+  PerunTranslateService,
+  StoreService,
+} from '@perun-web-apps/perun/services';
+import { MembersManagerService, RichMember, RichUser, User } from '@perun-web-apps/perun/openapi';
+
+export interface SponsorThisMemberDialogData {
+  theme: string;
+  member: RichMember;
+  sponsors: RichUser[];
+}
+
+@Component({
+  selector: 'app-sponsor-this-member-dialog',
+  templateUrl: './sponsor-this-member-dialog.components.html',
+  styleUrls: ['./sponsor-this-member-dialog.components.scss'],
+})
+export class SponsorThisMemberDialogComponent implements OnInit {
+  loading = false;
+  theme: string;
+  expiration = 'never';
+  selectedSponsor: User = null;
+  sponsorType = 'self';
+  member: RichMember;
+  sponsors: RichUser[];
+  minDate = new Date();
+
+  constructor(
+    private dialogRef: MatDialogRef<SponsorThisMemberDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: SponsorThisMemberDialogData,
+    private store: StoreService,
+    private membersService: MembersManagerService,
+    private notificator: NotificatorService,
+    private translate: PerunTranslateService
+  ) {}
+
+  ngOnInit(): void {
+    this.theme = this.data.theme;
+    this.member = this.data.member;
+    this.sponsors = this.data.sponsors;
+  }
+
+  onCancel(): void {
+    this.dialogRef.close(false);
+  }
+
+  onSubmit(): void {
```

Comment:
I believe there was an effort to stop using the 'onAction()' notation. Not sure though, please ask @bodnara  or @HejdaJakub .

Reply:
I believe in the Angular best practices it is advised  that **Output** of components should not be prefixed with 'on' so it does not collide with vanilla js e.g. `onClick`.
 `<e (something)="onSomething()">`  or `<e (something)="something()">`  should be ok
 `<e (onSomething)="whatever()">` will probably not break anything but preferably don't use this notation